### DISCUSSION TITLE
Skip license key error: The skip.tools license was created with a version of the software that expired on 2024-10-01

### DISCUSSION THREAD

**Discussion by marcprux** [skiptools/skip#250] [Announcements]
An error with the Skip license key checking had surfaced as of October 1st, 2024, which causes the following error to be issued when building a Skip project:

> The skip.tools license was created with a version of the software that expired on 2024-10-01 00:00:00 +0000 and must be re-generated

We have fixed the problem in Skip 1.1.6. You can upgrade from Xcode using `File`/`Packages`/`Update to Latest Package Versions`, or by setting `1.1.6` as the version in your `Package.swift`.

We apologize for the bug.

**Comment by bryanaamot** [comment]
Updated but still get the same error.
`❯ skip upgrade`
`[✓] Check Skip Updates: 1.1.7`
`[✓] Upgrade Skip to 1.1.7 (25.3s)`
`❯ skip version`
`Skip version 1.1.7`

Tried: Quitting Xcode. Cleaned Build folder. Deleting DerivedData folder. Running "swift build" from project root. Updating license key. Updating packages.

Here are my SPMs:
skip 0.10.11
skip-ffi 0.3.3
skip-foundation 0.7.4
skip-lib 0.8.3
skip-model 0.8.2
skip-ui 0.12.6
skip-unit 0.8.2

Running Xcode 15.4 on MacOS 14.4.1.

Got a pretty sweet project going with some automatic skip and some high speed platform specific Canvas graphics. Would like to continue if possible.

**Reply by aabewhite** [reply]
Your Package.swift hasn't been updated to Skip 1.0+. Unfortunately because 1.0 was a major version update, you have to manually edit Package.swift. See: https://skip.tools/docs/update-guide/#skip1_0migration and let us know if you have any issues!

**Reply by bryanaamot** [reply]
Thank you! That worked. All skip versions above 1.00 now except skip-ffi is at 0.41.
